### PR TITLE
Add property prefetch count unit tests for eh binder config

### PR DESCRIPTION
**Pull request by moarychan**
As title.

### CODE REVIEW

**Review by saragluna — APPROVED** [review]
LGTM.

File: sdk/spring/spring-cloud-azure-service/src/test/java/com/azure/spring/service/implementation/eventhubs/EventProcessorClientBuilderFactoryTests.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    @Override
+    protected TestAzureEventHubsProperties createMinimalServiceProperties() {
+        return new TestAzureEventHubsProperties();
+    }
+
+    @Test
+    void customPrefetchCount() {
+        TestAzureEventHubsProperties properties = createMinimalServiceProperties();
+        properties.getProcessor().setPrefetchCount(150);
+        final TestEventProcessorClientBuilderFactory builderFactory = new TestEventProcessorClientBuilderFactory(properties);
```
Comment (jialigit) [inline sdk/spring/spring-cloud-azure-service/src/test/java/com/azure/spring/service/implementation/eventhubs/EventProcessorClientBuilderFactoryTests.java:28]: why not spy the target object directly?

File: sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/config/EventHubsBinderConfigurationTests.java [hunk 2]
```diff
@@ -63,4 +95,146 @@ void shouldConfigureArmChannelProvisionerWhenResourceManagerProvided() {
...
+        new ApplicationContextRunner()
+            .withConfiguration(AutoConfigurations.of(TestProcessorContainerConfiguration.class))
+            .withPropertyValues(
+                "spring.cloud.stream.eventhubs.bindings.consume-in-0.consumer.prefetch-count=150",
+                "spring.cloud.stream.eventhubs.bindings.consume-in-0.consumer.destination=dest",
+                "spring.cloud.stream.eventhubs.bindings.consume-in-0.consumer.group=group",
+                "spring.cloud.stream.eventhubs.namespace=namespace"
+                )
+            .run(context -> {
+                EventHubsExtendedBindingProperties properties = context.getBean(EventHubsExtendedBindingProperties.class);
```
Comment (jialigit) [inline sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/config/EventHubsBinderConfigurationTests.java:110]: so  complicated?

File: sdk/spring/spring-cloud-azure-service/src/test/java/com/azure/spring/service/implementation/eventhubs/EventProcessorClientBuilderFactoryTests.java (hunk 1 above)
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-service/src/test/java/com/azure/spring/service/implementation/eventhubs/EventProcessorClientBuilderFactoryTests.java:28]: This method `prefetchCount` is our target to verify, then the builder `EventProcessorClientBuilder` should be a mock object. If using `spy` the factory class, then the real builder will be created.

File: sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/config/EventHubsBinderConfigurationTests.java (hunk 2 above)
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/config/EventHubsBinderConfigurationTests.java:110]: Yes, too complex, I don't find a better way to replace the mock builder.

**Pull request merged**